WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadViewModel.kt

```diff
@@ -1,15 +1,130 @@
+import org.wordpress.android.util.wizard.WizardState
+import org.wordpress.android.viewmodel.SingleEventObservable
+import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 
-class BackupDownloadViewModel @Inject constructor() : ViewModel() {
-    private var isStarted: Boolean = false
-    fun start() {
-        if (isStarted) {
-            return
-        }
+const val KEY_BACKUP_DOWNLOAD_ACTIVITY_ID_KEY: String = "key_backup_download_activity_id_key"
+const val KEY_BACKUP_DOWNLOAD_CURRENT_STEP = "key_backup_download_current_step"
+const val KEY_BACKUP_DOWNLOAD_COMPLETED = "key_backup_download_completed"
+const val KEY_BACKUP_DOWNLOAD_STATE = "key_backup_download_state"
+
+@Parcelize
+@SuppressLint("ParcelCreator")
+data class BackupDownloadState(
+    val siteId: Long? = null,
+    val rewindId: Long? = null
+) : WizardState, Parcelable
+
+typealias NavigationTarget = WizardNavigationTarget<BackupDownloadStep, BackupDownloadState>
+
+class BackupDownloadViewModel @Inject constructor(
+    private val wizardManager: WizardManager<BackupDownloadStep>
+) : ViewModel() {
+    private var isStarted = false
+    private var backupDownloadCompleted = false
+
+    private lateinit var backupDownloadState: BackupDownloadState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    clearOldBackupDownloadState(it)
+                    WizardNavigationTarget(it, backupDownloadState)
+                }
+        )
+    }
 
+    private val _wizardFinishedObservable = SingleLiveEvent<BackupDownloadWizardState>()
+    val wizardFinishedObservable: LiveData<BackupDownloadWizardState> = _wizardFinishedObservable
+
+    private val _screenTitleObservable = SingleLiveEvent<@StringRes Int>()
+    val screenTitleObservable: LiveData<Int> = _screenTitleObservable
+
+    private val _exitFlowObservable = SingleLiveEvent<Unit>()
+    val exitFlowObservable: LiveData<Unit> = _exitFlowObservable
+
+    private val _onBackPressedObservable = SingleLiveEvent<Unit>()
+    val onBackPressedObservable: LiveData<Unit> = _onBackPressedObservable
+
+    fun start(savedInstanceState: Bundle?) {
+        if (isStarted) return
+        if (savedInstanceState == null) {
+            backupDownloadState = BackupDownloadState()
+        } else {
+            backupDownloadCompleted = savedInstanceState.getBoolean(
+                    KEY_BACKUP_DOWNLOAD_COMPLETED,
+                    false
+            )
+            backupDownloadState = requireNotNull(
+                    savedInstanceState.getParcelable(
+                            KEY_BACKUP_DOWNLOAD_STATE
+                    )
+            )
+            val currentStepIndex = savedInstanceState.getInt(KEY_BACKUP_DOWNLOAD_CURRENT_STEP)
+            wizardManager.setCurrentStepIndex(currentStepIndex)
+        }
         isStarted = true
```

Comment:
Question (❓ ): Was this `isStarted` moved in between the if statement logic for a reason? If not, consider moving that close to the `if (isStarted) return` logic.

Reply:
> Was this isStarted moved in between the if statement logic for a reason? If not, consider moving that close to the if (isStarted) return logic.

Nope, I'll move it in a later branch

Reply:
in ef12c38